### PR TITLE
[SPLAT-1074] vSphere add template field for ipi zones

### PR DESCRIPTION
**Pull request by WenXinWei**
support template field in ipi zones.

### CODE REVIEW

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-f14

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-f14

**Comment by WenXinWei** [comment]
@jianlinliu @jinyunma @gpei please help review the pr when you have time. cc @jhou1  and @liangxia

File: ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh [hunk 1]
```diff
@@ -73,6 +75,7 @@ platform:
         networks:
         - ${LEASED_RESOURCE}
         datastore: mdcnc-ds-1
+        template: /${vsphere_datacenter}/vm/${vm_template}
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh:78]: If not setting the `template` value, what would happen?

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh:78]: Does it support on older version?

File: ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh [hunk 2]
```diff
@@ -24,6 +24,8 @@ mapfile -t vips < "${SHARED_DIR}/vips.txt"
 CONFIG="${SHARED_DIR}/install-config.yaml"
 base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
 machine_cidr=$(<"${SHARED_DIR}"/machinecidr.txt)
+ova_url=$(openshift-install coreos print-stream-json | jq -r '.architectures.x86_64.artifacts.vmware.formats.ova.disk.location')
+vm_template="${ova_url##*/}"
```
Comment (gpei) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh:28]: As I know, the output of `openshift-install coreos print-stream-json` may change from version to version, so suggest to add a validation check on the value of `ova_url` here to make sure it's a valid one.

Comment (gpei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh:28]: nevermind, if the output changes, the `jq` command should fail directly and the script aborted.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh [hunk 3]
```diff
@@ -24,6 +24,8 @@ mapfile -t vips < "${SHARED_DIR}/vips.txt"
 CONFIG="${SHARED_DIR}/install-config.yaml"
 base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
 machine_cidr=$(<"${SHARED_DIR}"/machinecidr.txt)
+ova_url=$(openshift-install coreos print-stream-json | jq -r '.architectures.x86_64.artifacts.vmware.formats.ova.disk.location')
```
Comment (jinyunma) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/ipi-conf-vsphere-zones-commands.sh:27]: It seems that we override the default behavior if appending field `template` directly in this zone configure step? Is it on purpose? 

And this step should be available on 4.12+, but field `template` starts to be supported on 4.14, so job might be failed on old version?

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-enable-template-f14

**Pull request closed without merging**